Tests/MongoSwiftTests/ServerSelectionTests.swift

```diff
@@ -23,19 +27,44 @@ private enum OperationType: String, Decodable {
 }
 
 final class ServerSelectionTests: MongoSwiftTestCase {
+    fileprivate func runTests(_ tests: [(String, ServerSelectionTestFile)]) throws {
+        for (filename, test) in tests {
+            print("Running test from \(filename)...")
+
+            // Server selection assumes that no read preference is passed for write operations.
+            let readPreference = test.operation == .write ? nil : test.readPreference
+            let heartbeatFrequencyMS = test.heartbeatFrequencyMS ?? SDAMConstants.defaultHeartbeatFrequencyMS
+
+            let selectedServers: [ServerDescription]
+            do {
+                selectedServers = try test.topologyDescription.findSuitableServers(
+                    readPreference: readPreference,
+                    heartbeatFrequencyMS: heartbeatFrequencyMS
+                )
+            } catch where test.error != true {
+                throw error
+            } catch {
+                continue
```

Comment:
should we assert on the error type here, or does the type thrown vary across tests?

Reply:
this should always be an `IllegalArgumentError` regarding max staleness validity, added an assertion